probatus/missing_values/imputation.py

```diff
@@ -0,0 +1,384 @@
+        """
+        if self.random_state is not None:
+            np.random.seed(self.random_state)
+
+        # Place holder for results.
+        results = []
+
+        self.X, self.column_names = preprocess_data(
+            X, column_names=column_names, verbose=self.verbose
+        )
+        self.y = preprocess_labels(y, index=self.X.index, verbose=self.verbose)
+
+        # Identify categorical features.
+        X_cat = X.select_dtypes(include=["category", "object"])
+        categorical_columns = X_cat.columns.to_list()
+
+        # Identify the numeric columns.Numeric columns are all columns expect the categorical columns
+        
+        X_num = X.drop(columns=categorical_columns, inplace=False)
+        numeric_columns = X_num.columns.to_list()
+
+        for strategy in self.strategies:
+
+            numeric_transformer = Pipeline(
+                steps=[("imputer", self.strategies[strategy])]
+            )
+
+            categorical_transformer = Pipeline(
+                steps=[
+                    (
+                        "imp_cat",
+                        SimpleImputer(
+                            strategy="constant",
+                            fill_value="missing",
+                            add_indicator=True,
+                        ),
+                    ),
+                    ("ohe_cat", OneHotEncoder(handle_unknown="ignore")),
+                ]
+            )
+
+            preprocessor = ColumnTransformer(
+                transformers=[
+                    ("num", numeric_transformer, numeric_columns),
+                    ("cat", categorical_transformer, categorical_columns),
+                ],
+                remainder="passthrough",
+            )
+
+            model_pipeline = Pipeline(
+                steps=[("preprocessor", preprocessor), ("classifier", self.clf)]
+            )
+
+            temp_results = self._calculate_results(
+                X, y, clf=model_pipeline, strategy=strategy
+            )
+
+            results.append(temp_results)
+
+        # If model supports missing values by default, then calculate the scores
+        # on raw data without any imputation.
+        if self.model_na_support:
+
+            categorical_transformer = Pipeline(
+                steps=[
+                    ("ohe_cat", OneHotEncoder(handle_unknown="ignore")),
+                ]
+            )
+
+            preprocessor = ColumnTransformer(
+                transformers=[("cat", categorical_transformer, categorical_columns)],
+                remainder="passthrough",
+            )
+
+            model_pipeline = Pipeline(
+                steps=[("preprocessor", preprocessor), ("classifier", self.clf)]
+            )
+
+            temp_results = self._calculate_results(
+                X, y, clf=model_pipeline, strategy="No Imputation"
+            )
+            results.append(temp_results)
+
+        self.report_df = pd.DataFrame(results)
```

Comment:
Maybe set index of the df as the name of imputation method, and then the columns would be train score mean, train score std, val_score_mean, val_score_std. I think this way the report will be cleaner and have more information.

What do you think?

Reply:
Done.The output now has all the metrics the user can look through.